plutus-core/index-envs/src/Data/RandomAccessList/SkewBinary.hs

```diff
@@ -63,6 +64,7 @@ cons :: a -> RAList a -> RAList a
 cons x = \case
     (BHead w1 t1 (BHead w2 t2 ts')) | w1 == w2 -> BHead (2*w1+1) (Node x t1 t2) ts'
```

Comment:
Do you think using `unsafeShiftL` in place of `2*w1` multiplication is worth it? I have a feeling that GHC would already apply such transformation

Reply:
IIRC GHC isn't actually smart enough to do that, but I may be lying to you. But it's certainly best not to rely on GHC when it's easy.

Reply:
Just checked, yes, GHC doesn't optimize that one:

```
plusWord64# (timesWord64# 2#Word64 bx_danZ) 1#Word64
```

Maybe `timeWord64#` somehow does, but I kinda doubt.

@kwxm I'll try your version out.